fiftyone/core/session/session.py

```
@@ -1298,6 +1369,70 @@ def _on_refresh(session: Session, state: t.Optional[StateDescription]):
+    result = []
+    for item in samples:
+        if isinstance(item, str):
+            result.append({"sample_id": item, "type": "default"})
+        elif isinstance(item, dict):
+            sample_id = item.get("sample_id")
+            sel_type = item.get("type", "default")
+            if sel_type not in valid_types:
+                raise ValueError(
+                    f"Invalid selection type '{sel_type}' for sample "
+                    f"'{sample_id}'. Must be one of {valid_types}"
+                )
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)